tests/frontend/test_io.py

```diff
@@ -453,6 +459,65 @@ def test_invalid_compilation(self):
              prog = io.to_program(bb)
 
 
+class DummyResults:
+    """Dummy results object"""
+
+
+def dummy_run(self, program, compile_options=None, **eng_run_options):
+    """A dummy run function, that when called returns a dummy
+    results object, with run options available as an attribute.
+    This allows run_options to be returned and inspected after calling eng.run
+    """
+    results = DummyResults()
+    results.run_options = eng_run_options
+    return results
+
+
+class TestEngineIntegration:
```

Comment:
I guess this is a "frontend integration" test?

Reply:
I guess so. Frontend integration in that it is testing multiple parts of the frontend (io.load, Program, eng.run()). But everything is still contained within the frontend.